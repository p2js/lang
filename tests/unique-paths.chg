mod m1 {
    fn foo() {}
}
mod m2 {
    fn foo() {}
}
mod m3 {
    fn main() {}
}
fn main() {
    m1::foo();
    m2::foo();
    m3::main();
    main();
}
