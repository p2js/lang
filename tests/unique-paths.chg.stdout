expr @ 2:14 3:0 => ()
expr @ 5:14 6:0 => ()
expr @ 8:15 9:0 => ()
expr @ 11:9 11:12 => fn foo(...)
expr @ 11:9 11:14 => ()
expr @ 12:9 12:12 => fn foo(...)
expr @ 12:9 12:14 => ()
expr @ 13:9 13:13 => fn main(...)
expr @ 13:9 13:15 => ()
expr @ 14:5 14:9 => fn main(...)
expr @ 14:5 14:11 => ()
expr @ 10:11 16:1 => ()



define void @root$m1$foo() {
prologue:
br label %bb.0

bb.0:
ret void

}

define void @root$m2$foo() {
prologue:
br label %bb.0

bb.0:
ret void

}

define void @root$m3$main() {
prologue:
br label %bb.0

bb.0:
ret void

}

define void @main() {
prologue:
br label %bb.0

bb.0:
%l.0 = call {} bitcast(void()* @root$m1$foo to {}()*)()
br label %bb.1

bb.1:
%l.1 = call {} bitcast(void()* @root$m2$foo to {}()*)()
br label %bb.2

bb.2:
%l.2 = call {} bitcast(void()* @root$m3$main to {}()*)()
br label %bb.3

bb.3:
%l.3 = call {} bitcast(void()* @main to {}()*)()
br label %bb.4

bb.4:
ret void

}


